### DISCUSSION TITLE
WPF-based CustomTaskPane (CTP)

### DISCUSSION THREAD

**Discussion by ricardogerbaudo** [Excel-DNA/ExcelDna#504] [Q&A]
Quick question: in order to use a WPF UserControl within a CTP, the only way is to embed through a WinForms ElementHost class?

**Comment by govert** [comment]
Hi @ricardogerbaudo - the Office CTP mechanism requires an ActiveX control. My understanding is that WPF has no built-in support for exposing controls as ActiveX controls, while Windows Forms does have such support. The ElementHost control is the bridge that lets the WPF be embedded in a Windows Forms control, and hence exposed as an ActiveX control. So as far as I know, you can't get rid of the extra layer easily. I don't think the ElementHost / WinForms aspect has been a particular source of trouble.

I also note there was no answer to the same question on StackOverflow: https://stackoverflow.com/questions/56023503/hosting-wpf-content-directly-in-taskpane
